catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/java/org/codice/alliance/catalog/plugin/defaultsecurity/DefaultSecurityAttributeValuesPlugin.java

```diff
@@ -23,94 +25,104 @@
+  private static final String OTHER_DISSEMINATION_CONTROLS_CONFIGURATION_KEY =
+      "otherDisseminationControls";
 
-  static final String DISSEMINATION_CONTROLS_KEY = "disseminationControls";
+  private static final String OWNER_PRODUCER_CONFIGURATION_KEY = "ownerProducer";
 
-  static final String OTHER_DISSEMINATION_CONTROLS_KEY = "otherDisseminationControls";
+  private static final String DEFAULT_MARKINGS_TAG = "defaultMarkings";
 
-  static final String OWNER_PRODUCER_KEY = "ownerProducer";
+  private final SecurityAttributes securityAttributes;
 
-  private static final Map<String, Set<String>> SYS_HIGH_TO_METACARD_ATTRIBUTE_MAPPING =
-      new HashMap<>();
+  private final SystemHighAttributes systemHighAttributes;
 
-  public DefaultSecurityAttributeValuesPlugin(
-      SecurityAttributes securityAttributes, Map<String, Object> initialValues) {
-    this(
-        securityAttributes,
-        initialValues,
-        () -> org.codice.ddf.security.common.Security.getInstance().getSystemSubject());
-  }
+  /**
+   * This map will contain an entry for all 6 of the metacard attribute names. It is {@code
+   * volatile} and not synchronized because it could be re-assigned in {@link
+   * DefaultSecurityAttributeValuesPlugin#update(Map)} while being iterated through in {@link
+   * DefaultSecurityAttributeValuesPlugin#process(CreateRequest)}.
+   */
+  private volatile Map<String, String> metacardAttributeNameToSystemHighAttributeNameMap;
 
   public DefaultSecurityAttributeValuesPlugin(
       SecurityAttributes securityAttributes,
-      Map<String, Object> initialValues,
-      Supplier<Subject> subjectSupplier) {
-    this.securityAttributes = securityAttributes;
-    this.subjectSupplier = subjectSupplier;
-
+      SystemHighAttributes systemHighAttributes,
+      Map<String, String> initialValues) {
+    this.securityAttributes = notNull(securityAttributes, "SecurityAttributes may not be null");
+    this.systemHighAttributes =
+        notNull(systemHighAttributes, "SystemHighAttributes may not be null");
     update(initialValues);
   }
 
-  public void update(Map<String, Object> properties) {
-    if (properties != null && !properties.isEmpty()) {
-      setClassification((String) properties.get(CLASSIFICATION_KEY));
-      setReleasability((String) properties.get(RELEASABILITY_KEY));
-      setCodewords((String) properties.get(CODEWORDS_KEY));
-      setDisseminationControls((String) properties.get(DISSEMINATION_CONTROLS_KEY));
-      setOtherDisseminationControls((String) properties.get(OTHER_DISSEMINATION_CONTROLS_KEY));
-      setOwnerProducer((String) properties.get(OWNER_PRODUCER_KEY));
-    }
+  /** Updates cached default system high attribute values. */
+  public void update(Map<String, String> properties) {
+    notEmpty(properties, "properties may not be empty");
+
+    final Map<String, String> newMetacardAttributeNameToConfigurationValueMap = new HashMap<>();
+
+    newMetacardAttributeNameToConfigurationValueMap.put(
```

Comment:
❓ https://github.com/codice/alliance/pull/453#discussion_r153341413?

Reply:
In this case, I'd like to throw an exception and just fail the update without logging anything. I think we can count on blueprint to never allow these values to be empty, so it shouldn't ever come up.